syntax = "proto3";

import "Texture.proto";
import "Program.proto";
import "Scene.proto";
import "Material.proto";

package frame.proto;

// Level this describe the level loading of the app.
// Next 9
message Level {
	// Level name.
	string name = 1;
	// default texture name, this is the texture that will be shown.
	string default_texture_name = 2;
	// Contains the needed textures.
	repeated Texture textures = 5;
	// Contains the needed programs.
	repeated Program programs = 6;
	// Contains the scene tree.
	SceneTree scene_tree = 7;
	// Contains the needed materials.
	repeated Material materials = 8;
}
